Tensile/KernelWriterAssembly.py

```diff
@@ -1568,6 +1571,38 @@ def defineMACMacro(self, kernel, innerUnroll):
             else:
               printExit("Half-precision not supported for arch=%u" % self.version )
 
+                  cStr = "v[%s+%u*2+%u*%u*2+0*2+0]" % ("vgprValuC", blockA, blockB, kernel["ThreadTile0"]) # *2 b/c of fp32
+                  cidx = blockA*2 + blockB*kernel["ThreadTile0"]*2 + 0
+
+      # integer i8
+      elif kernel["ProblemType"]["DataType"].isInt8x4():
+        for b in range(0, kernel["ThreadTile1"]):
+          for a in range(0, kernel["ThreadTile0"]):
+            if self.version == (8,0,3):
+              kStr += self.comment3("int8 not implemented yet for gfx803:")
+            elif self.version == (9,0,0):
+              kStr += self.comment3("int8 not implemented yet for gfx900:")
+            elif self.version == (9,0,6):
+              for iui in range(0, innerUnroll):
+                cidx = a + b*kernel["ThreadTile0"] + 0
+                cStr = "v[%s+%u+%u*%u]" % ("vgprValuC", a, b, kernel["ThreadTile0"])
+                aStr = "v[%s+%u]"       % ("vgprValuA_X%u_I%u"%(m,iui), a)
+                bStr = "v[%s+%u]"       % ("vgprValuB_X%u_I%u"%(m,iui), b)
+# this will be same as "single precision" except for the dot4 instruction replacing the v_mac_f32 instruction
+# below is a guss of the v_dot_i32_i8 instruction based on the dot2 instruction. It will need editing to account 
+# for the registers for a and b having 4 values, and the need to add saturation directive to clamp integer arithmetic
+//              kStr += "v_mac_f32 %s, %s, %s%s" % (cStr, aStr, bStr, self.endLine) # single precision
+//              kStr += "v_dot2_f32_f16 %s, %s, %s, %s op_sel:[0,0] op_sel_hi:[1,1] //valuC[%u]%s" % (cStr, aStr, bStr, cStr, cidx, self.endLine) # hpa_hgemm
+                kStr += "v_dot4_i32_i8  %s, %s, %s, %s op_sel:[0,0] op_sel_hi:[1,1] //valuC[%u]%s" % (cStr, aStr, bStr, cStr, cidx, self.endLine)
```

Comment:
Above use of v_dot4_i32_i8 is a guess. Is there a good reference for this command? I expect there may be 4 op_sel directives for the 4 i8 values. I also expect there is a directive to saturation directive to clamping